lib/model/port.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * port.js
+ *
+ * Definition for the Port class. Ports are the points on a Node where
+ * connections attach to or from the node. They can be either inputs
+ * or outputs.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Port's unique identifier.
+ * @param type - The type of port. Valid options include "input" or "output".
+ * @param direction - cardinal direction (N,S,E,W) in which the port is located on the node.

```

Comment:
Just one port. I'm proposing having `Port` compose `PortPosition` which would contain the `direction` and `percentage` properties.